include/swift/AST/TypeCheckRequests.h

```diff
@@ -2419,15 +2443,17 @@ class ResolveTypeRequest
   // Cycle handling.
   void noteCycleStep(DiagnosticEngine &diags) const;
 
+  // Source location
+  SourceLoc getNearestLoc() const;
+
 private:
   friend SimpleRequest;
 
   // Evaluation.
-  Type evaluate(Evaluator &evaluator, TypeResolution *resolution,
-                TypeRepr *repr) const;
+  Type evaluate(Evaluator &evaluator,
+                TypeResolutionDescriptor descriptor) const;
```

Comment:
Why the upgrade to a descriptor?

Reply:
This was left over from when I was building on top of `TypeResolution`. I only kept it because of the nice `simple_display` output, but I don't mind downgrading.